app/vmselect/main.go

```diff
@@ -124,6 +124,11 @@ func main() {
 		if err != nil {
 			logger.Fatalf("cannot initialize vmselectapi server: %s", err)
 		}
+
+		if len(*storageNodes) < netstorage.GetReplicationFactor() {
+			logger.Warnf("number of replicationFactor should less than number of storageNode.")			
```

Comment:
This should be a fatal error, which stops further execution of `vmselect`. Otherwise `vmselect` will continue running with the incorrect config, which may lead to incorrect query results.

Also it is unclear how this check should interact with [the dynamic discovery of `vmstorage` nodes](https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#automatic-vmstorage-discovery). Sometimes the number of discovered `vmstorage` nodes may become smaller than the specified `-replicationFactor`. For example, during the upgrade of `vmstorage` nodes via [minimum downtime strategy](https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#minimum-downtime-strategy).

See also https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3805#issuecomment-1438217315